spec/std/channel_spec.cr

```diff
@@ -138,27 +157,44 @@ describe Channel::Unbuffered do
     ch.receive?.should eq(123)
   end
 
-  it "wakes up the sender fiber when channel is closed" do
+  it "wakes up sender fiber when channel is closed" do
     ch = Channel::Unbuffered(Nil).new
-    sender_closed = false
-    spawn do
-      ch.send nil
-      ch.send nil
-    rescue Channel::ClosedError
-      sender_closed = true
+    closed = false
+    main = Fiber.current
+
+    sender = Fiber.new do
+      ch.send(nil)
```

Comment:
Related to #6948 : I'd expect `ch.send` to raise here since it couldn't deliver the value.